spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/reload/Fabric8EventBasedSecretsChangeDetector.java

```diff
@@ -105,13 +111,13 @@ private void inform() {
 	}
 
 	protected void onEvent(Secret secret) {
-		log.debug("onEvent secrets: " + secret.toString());
-		boolean changed = changed(locateMapPropertySources(fabric8SecretsPropertySourceLocator, environment),
-				findPropertySources(Fabric8SecretsPropertySource.class));
-		if (changed) {
-			log.info("Detected change in secrets");
+
+		boolean reload = ConfigReloadUtil.reload("secrets", secret.toString(), fabric8SecretsPropertySourceLocator,
```

Comment:
you can think about this as the core change. We do this logic in 4 places, just that the classes are different, but its exactly the same. So I decided to move it to common